# Introduction
**Investigate a Dataset-Titanic Project**

Hello, my name is Tae Hyun Yun. As a part of the Udacity Data Analyst program,
I am going to use the Titanic data set provided at [Kaggle](https://www.kaggle.com/c/titanic/data) to answer what could have caused the deaths that happened on that day.

# What affected their survival rate the most?

Like all other scientific studies, a hypothesis for this test seems fitting.
My hypothesis is that: **Gender, age, and Pclass will have the largest impact on the people's social standing in that order.**
To do this I want to use the pandas module and the numpy module and will load it in code below:

In [1]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt 
import scipy
%matplotlib inline  

# Subset


In [101]:
df = pd.read_csv('train.csv')
#lets subset
lp=df[df['Survived']==1]
dp=df[df['Survived']==0]
agelp=lp[np.isfinite(lp['Age'])]

# Age Distribution of Survivors

In [94]:
lp.Age.describe()

C:\Users\Tae Hyun Yun\OneDrive\Documents\Spring 2017\dATA ANALYST\Project 0\Anaconda3\lib\site-packages\numpy\lib\function_base.py:3834: RuntimeWarning: Invalid value encountered in percentile
  RuntimeWarning)


count    290.000000
mean      28.343690
std       14.950952
min        0.420000
25%             NaN
50%             NaN
75%             NaN
max       80.000000
Name: Age, dtype: float64

The mean shows 28.343.

# Age Distribution of the Dead

In [95]:
dp.Age.describe()

C:\Users\Tae Hyun Yun\OneDrive\Documents\Spring 2017\dATA ANALYST\Project 0\Anaconda3\lib\site-packages\numpy\lib\function_base.py:3834: RuntimeWarning: Invalid value encountered in percentile
  RuntimeWarning)


count    424.000000
mean      30.626179
std       14.172110
min        1.000000
25%             NaN
50%             NaN
75%             NaN
max       74.000000
Name: Age, dtype: float64

The mean shows 30, higher than the survivors.

# Passenger Class 
**Of Survivors**

In [96]:
lp.Pclass.describe()

count    342.000000
mean       1.950292
std        0.863321
min        1.000000
25%        1.000000
50%        2.000000
75%        3.000000
max        3.000000
Name: Pclass, dtype: float64

**Of the Dead**

In [97]:
dp.Pclass.describe()

count    549.000000
mean       2.531876
std        0.735805
min        1.000000
25%        2.000000
50%        3.000000
75%        3.000000
max        3.000000
Name: Pclass, dtype: float64

It looks like there are high number of Passenger 3 class number in death but evenly distributed for survivor class.

# Gender Distribution

In [99]:
lp.Sex.describe()

count        342
unique         2
top       female
freq         233
Name: Sex, dtype: object

233 out of 342 were female an overwhelming difference compared to the male number of 109.

# Prediction machine

Based on the statistical description study above, it was noted that Age, Passenger Class, and Sex ratio had a strong correlation with survival rate. To test this and understand the weight of each of these factors, I ran a manual weighting prediction machine below to see which factors actually impacted the survival rate.

In [188]:
### prediction function based on rough estimates of weight which is based on the mean and the standard deviations
def prediction(i):
    score=0
    if abs(lp.Age.mean()-df.Age[i]) < abs(dp.Age.mean()-df.Age[i]):
        score+=0
    elif abs(lp.Age.mean()-df.Age[i]) > abs(dp.Age.mean()-df.Age[i]):
        score+=-0.1
    if abs(lp.Pclass.mean()-df.Pclass[i])<abs(dp.Pclass.mean()-df.Pclass[i]):
        score+=0.5
    elif abs(lp.Pclass.mean()-df.Pclass[i])>abs(dp.Pclass.mean()-df.Pclass[i]):
        score+=-.2
    if df.Sex[i]=="female":
        score+=0.3
    elif df.Sex[i]=="male":
        score+=-0.9
    return score
def accuracy(i,score):
    if score>0 and df.Survived[i]==1:
        boo=1
    elif score<0 and df.Survived[i]==0:
        boo=1
    else:
        boo=0
    return boo

percentage=[]
for i in df.PassengerId:
    percentage.append(accuracy(i-1,prediction(i-1)))
ans=sum(percentage)/len(percentage)
print(ans)


0.797979797979798


# Revised Prediction Machine

I switched up the conditions to improve the prediction rate. They were based on mean and mode of the statistical study result

In [220]:
### prediction function based on rough estimates of weight which is based on the mean and the standard deviations
def prediction(i):
    score=0
    if df.Age[i]<28.38:
        score+=0.15
    elif df.Age[i]>=28.38:
        score+=-0.15
    if df.Pclass[i]==1:
        score+=.2     
    elif df.Pclass[i]==3:
        score+=-.2
    if df.Sex[i]=="female":
        score+=0.3
    elif df.Sex[i]=="male":
        score+=-0.2
    return score
def accuracy(i,score):
    if score>0 and df.Survived[i]==1:
        boo=1
    elif score<=0 and df.Survived[i]==0:
        boo=1
    else:
        boo=0
    return boo

percentage=[]
for i in df.PassengerId:
    percentage.append(accuracy(i-1,prediction(i-1)))
bro=sum(percentage)/len(percentage)
print(bro)

0.8024691358024691


The gender of a given person seems to play a bigger role in determining a factor of survival than passenger class or age in that respective order of importance.

# Conclusions

From this we can learn, also with the background of the Titanic Story, that Women, Children, and 1st Class passengers had priorities to the boat with differing levels of priority. We can also learn that data analytics can speculate and understand from just data which is very powerful in any context. 